src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -95,6 +95,9 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
             this.commandManager.registerCommand(DSCommands.DebugNotebook, async () => {
                 const editor = this.vscNotebook.activeNotebookEditor;
                 if (editor) {
+                    if (this.notebookToDebugger.has(editor.document)) {
```

Comment:
Not sure this is going to be enough.
If user clicks this button while `checkforIpykernel6` is still running, then yo'ull end up with duplicate sessions.

Reply:
I think you can move this entire if condition from line 98 to 109 into a seprate fucntion, this has been repeated in 3 places, and then you can easily check if that method has already been called once before if `yes`, then return.